### DISCUSSION TITLE
CTRLR Workflows vs Versions ( and URLS to various Versions )

### DISCUSSION THREAD

**Discussion by unityconstruct** [RomanKubiak/ctrlr#601] [Q&A] [answered]
_Version Table that's possibly deprecated/incorrect at time of reading_

| Feature | Operation | mac | pc | Comment |
| -- | -- | -- | -- | -- |
| VST3 | PANEL DEV | `5.6.23+`  | `5.6.23+`  |  |
| VST3 | PANEL EXPORT | `5.6.23+`  | `5.6.23+` | VST3 builds require at least the Ctrlr version moddified by @jmechnich to work properly with xcode. <br/> https://github.com/jmechnich/ctrlr/releases/tag/5.6-f452347 |
| VST2 | PANEL DEV | PC: `5.3.201` | MAC: `5.3.198` |  |
| VST2 | PANEL EXPORT | PC: `5.3.201` | MAC: `5.3.198` |  |
| AU | PANEL DEV | PC: `5.3.201` | MAC: `5.3.198` |  |
| AU | PANEL EXPORT | PC: `5.3.201` | MAC: `5.3.198` |  |
| STANDALONE | PANEL EXPORT | PC: `5.3.201` | MAC: `5.3.198` |  |
| STANDALONE | PANEL EXPORT | PC: `5.3.201` | MAC: `5.3.198` |  |
| CTRLR | DEV | `CtrlrX-5.6.29` | `CtrlrX-5.6.29` |  |
|  | PANEL |  |  |  |

- My takeways
  - VST3: PANEL DEV: `6.0.26` _(to pick up JUCE or other enhancements + stability.. BUT, no MIDI(?))_
    - requires project rebuild to pick up VST3 instrument names, etc
  - VST3: PANEL EXPORT: `Ctrlr-5.6.23`,`CtrlrX-5.6.29`
  - VST2: PANEL DEV: PC: `5.3.201` | MAC: `5.3.198` 
  - VST2: PANEL EXPORT: PC: `5.3.201` | MAC: `5.3.198`
  - AU: PANEL DEV: `5.3.201` | MAC: `5.3.198`
  - AU: PANEL EXPORT: `5.3.201` | MAC: `5.3.198`
  - STANDALONE: PANEL DEV: `5.3.201` | MAC: `5.3.198`
  - STANDALONE: PANEL EXPORT: `5.3.201` | MAC: `5.3.198`
  - CTRLR DEV: Latest development work is at CtrlrX-`5.6.29`

### Old-Old Official Builds_(ctrlr.org)_
- https://ctrlr.org/nightly/?C=M;O=D
### New-Old Official Releases _(github)_
- https://github.com/RomanKubiak/ctrlr/releases

<hr/>

### Current Tip Revisions [as of 2024-01-29] :
- ctrlr - `5.6.0`= `5.6.23`
- CtrlrX - `5.6.29` _(forked from `5.6.23`)_ _( master merge a few days ago )_

### Ctrlr Versions

_table under dev_

| Repo | Version | Lua | JUCE | Features | Comment |
| -- | -- | -- | -- | -- | -- |
| Ctrlr | `5.3.198` | ???? | ????? | VST2.4,STANDALONE | for MAC |
| Ctrlr | `5.3.201` | ????,| ????? | VST2.4,STANDALONE | for PC |
| Ctrlr | `5.6.0` | `5.1.4` | `6.0.8` |  |  |
| Ctrlr | `5.6.0` | `5.1.4` | `6.0.8` |  |  |
| Ctrlr | `5.6.23` | `5.1.4` | `6.0.8` |  |  |
| CtrlrX | `5.6.29` | `5.1.4` | `6.0.8` |  |  |
| Ctrlr | `6.0.26` | `5.1.4` | `6.0.8` |  |  |

**Comment by unityconstruct** [comment]
RE: CTRLR DEV
### I imagine there are at least 2 'active' tips various peeps are on: 
- 5.6+: for VST3/JUCE6.0.8, MIDI support disabled during the migration
- <5.6: for everything else
- CtrlrX seemed to have most recent updates
- 6.x: no idea where this is, what features are involved, or known issues rendering it tailored to a specific purpose.

**Reply by Godlike-Productions** [reply]
Thanks for all the efforts here.  

One thing that would be good to integrate into the v5.6.29 development would be the improved error handling of LUA methods that is in 6.0.26 and 5.3.201.  I have no idea what needs to be done here.

Also, if we get a release that we're all happy with, I'm happy to host it over at godlike.com.au as I have done previously, as well as any documentation.

**Reply by unityconstruct** [reply]
My personal TODO/Wiki is here: https://github.com/unityconstruct/CtrlrXY/wiki

Until I'm done with that list, trying to stay out of the way & happy to help out where I can.... hopefully i the realm of the above mentioned technical stuffs. I'm as hack-tastic as they come & prolific with note taking - in no small part b/c my hobbies far exceed my ability to manage them all.  :)

Parsing the boards astonishes me with many of the design patterns, beautiful UIs, and craftsmenship... hat's off to whomever for somehow managing to convince Roman not to flush it all.

**Reply by bijlevel** [reply]
Nice idea, the CTRLRXY/wiki. I did visit it and you may add my name as the author of the guide about compiling for Windows 1O: **Sjoerd W Bijleveld**. Also there is a guide for the Mac: https://www.godlike.com.au/fileadmin/godlike/techtools/ctrlr/guides/My_guide_to_compiling_Ctrlr_for_macOS__Mojave__v2.pdf from the same author ;-). I think those guides also apply to higher versions of Windows and MacOS (Intel and Silicon).

**Reply by Godlike-Productions** [reply]
@unityconstruct Again thanks so much for your time.  I wish I knew more C++ stuff and could help.  I'm slowly working my way through a course, but I'm a beginner with C++ and Juce.

> a side benefit to this ( i think ) is that I believe once that's done, one can develop Lua scripts IN visual-studio & get intellisense help & hopefully headoff exceptions before they happen AND let the IDE to you WTF it wants before you do you it 

I used to do this with 6.0.26.  You can link to LUA files rather than have them hosted within ctrlr.  This allowed me to maintain a git repository of the code changes while I was building Jupiter 8 Morph, and I could edit them with VS and push the changes to the repo.  It was very convenient.  I think you can still do it with 5.6.23 but from memory the export function wasn't supported, like it was in 6.0.26.

I did document it in the CtrlrX wiki here - https://github.com/damiensellier/CtrlrX/wiki/Exporting-code-for-repositories

Might be a shortcut for you.

Again, thankyou so much for sharing your expertise here.  It has been sorely needed.  We are predominantly sysex freaks and musicians flailing around in Juce as best we can and helping each other out as we get better at stuff.

**Reply by damiensellier** [reply]
@unityconstruct I think the only limitation to upgrade to JUCE 7 or higher is that LUABIND is deprecated and has the old C++ version. The way to improve the LUA side of Ctrlr would be to replace all the binding with another LUA<->C++  library like [Luaaa](https://github.com/gengyong/luaaa) or [LuaBridge3](https://github.com/kunitoki/LuaBridge3). There's other binders of course.

**Reply by unityconstruct** [reply]
### CtrlrXY Wiki updated
- Wiki Updated at: https://github.com/unityconstruct/CtrlrXY/wiki

### CtrlrDocs
- No sooner than I got started, the Wiki limitations became quite obvious, so leaving the guide/release pages up there, but created a NEW Repo CtrlrDocs that is 100% separate from the code.
- Considering that 60-lesson zip file will entail 60 pages, the wiki will become a hot mess befor end-of-month :)
- By leveraging `*.md` markdown files (with an onboard `img` to supply graphics), CtrlrDocs repo will basically be a web site, with each asset having browser rendering support.
- I really don't want to diverge off CtrlrX - or cause mass confusion... or accidentally kick off PRs to CtrlrX farting around _(never happened)_ , my main intention with that repo is have something I can play around it without mucking this repo or damien's ( any more than I already have )

### .gitmodules
```gitmodules
[submodule "Panels"]
	path = Panels
	url = https://github.com/RomanKubiak/Panels.git

[submodule "CtrlrDocs"]
	path = CtrlrDocs
	url = https://github.com/unityconstruct/CtrlrDocs.git
```

#### Example Page
![image](https://github.com/RomanKubiak/ctrlr/assets/40268430/e7918e16-966b-461e-83c8-2e40e2c64029)

- Just like `Panels` is added to Ctrlt[X] through `.gitmodules` , this repo can be optionally added, yet remain completely independent & genereally accessible:
- With a more complex URL, specific branches/commits can be targeted/frozen to avoid having documentation been 50years old on the CtrlrX master - or the other way around.

- even better yet, this allows anyone to steal CtrlrDocs wholesale (or by forking) and optionally add/remove it to any Ctrlr project/fork with complete independence.

### LuaLib ( CtrlrLuaLib? maybe )
- So keeping all that in mind, seems that my Lua script aggregation can live far & away from Ctrl[X] to void same pitfalls.
- Perhaps sometime in the future this could be quite massive, well commented/documented, and be integrated to a Ctrlr repo/project
- Concepts from CtrlrDocs will make their way here & be categorized, organized.... I'm only on the 3rd blog and there's a ton of top-notch design patterns/principles.
- Once a library is well-formed, stable, and providing full coverage of the LuaLib bindings API, then the natural progression would be to wire-up the utility clases/methods as a framework (think SprintBoot) and this framework would be a safe place to sit panel design on top of... 
- As Ctrlr matures or changes, the framework should provide not only a very intuitive workflow for designers, but act like a nimble shock-absorber that can be re-wired on the back-end with out panels knowing - or caring.
- Theoretically, this would make for a good segue to embed it all in Ctrlr, possibly allowing most everything but data structures & transformation to be done without touching Lua, only needing to type in method names for listeners/callback & dropping in values.
- This is analogous to `... I give two turds about typecasting and markup for bolding/itatlics/cell color alignment... I just click a ton of buttons in 30min and have what I need - usually i don't even have to type anything to transform data(click data>import>select file>click delimeter>OK... done..
  - if I can't click my way to happiness, then Excel has a lightweight formula engine
  - if I can't do it with that, I have macros & a full C#/VB object set to get stupid with.
### .gitmodules
```gitmodules
[submodule "LuaLib"]
	path = LuaLib
	url = https://github.com/unityconstruct/LuaLib.git
```

### CtrlrLuaVendorLib
- so I'm forcing myself to stop dreaming up endless possibilities, before I spew another novel twice as long as the last, but...
- Sure seems handy someday to have the vendor/model sysex trapped in data strucs that are fed to Lua Object Factories to produce Presets/Dumps from our lovely Excel analogy such that one could probably come damn close to doing most everything without too much coding... Like Excel itself
- Where a vendor reuses specs across models, new instrument support seems more approachable..

- I'm an SDET professionally, so my life is largely dedicated to wrapping APIs to enable succinct maintainable Test code where doing more with less is valued. So this is where all this is coming from conceptually
- That's the good news... the bad is I don't know the diff between C/C++ & a hole in the ground... and syntactically, just looks like my cat run up & down my keyboard ( but I'll manage!)


ok shutting up now before I think of something else I'll probably never get around to

Cheers!

**Reply by unityconstruct** [reply]
@Godlike-Productions ... lol... if you knew how lost I am looking at the noodlefest they call C

BUT.. looking at those zipped blog archives.. there's a shittone of awesome stuff, from both UI design and programming principles... I'm somewhat weary at jumping into the code too much - I don't konw how many languages I know enough of to get the damn job done & God has a sense of humor I haven't figured out yet b/c seems like every other contract bounces between Java & C# so I'm nimble but NEVER get the luxury of getting mystical with either.

**Reply by unityconstruct** [reply]
@damiensellier ... yeah I've moved on from that idea  :-P  `JUCE not worth the squeeze....` haha.. ok that was stupid.
- bug fix is way to go... like you said.. 
- there's so much information out there that I'll be busy for a long while simply saving it all off, much less pooping it out to MD
- and no sense in me f'ing things up.. by the time I get through all the tutorials and Lua trix... it might be safe...
- I don't mind triage & helping out, but trying to stay out of the way of CtrlrX ENTIRELY, opting for artifacts that can be brought i optionally.. once they don't suck... then I'll try to sell some integration plan.

Can't say enough about your recent facelift of CtrlrX brother!!!!! Didn't look too trivial.. & I hope that mapping some of all this out might help your or one of the other gurus. The shear volume of information seems to lend itself to forgetting kool stuff as soon as new kool stuff hits the desktop.

**Reply by Godlike-Productions** [reply]
Oh I wish SYSEX was so structured.  Its designed specifically for model specific support.  I think the only universal command is the model ID request, and even among manufacturers their SYSEX model varies wildly.  Many are not even publicly available, especially sysex dump formats.

By way of example, we've been reverse engineering the Kurzweil patch dump format for over 20 years, and finally have most of it mapped, but it's been a monumental effort.  A patch has thousands parameters, most of which are optional, and even with the support of Kurzweil it has been difficult, as it isn't documented there either.  So many companies simply no longer exist, so when developing a panel it's really a matter of researching the synth in question and mapping control to a message the synth understands.  Sometimes we've had to hack functionality that was never intended (such as my Jupiter 8 Morph Panel) as the synth never supported MIDI.

Heck, even a library of links to MIDI specs of synths would be amazing, let alone full blown support within the programming interfaces.

Companies like Soundtower, SoundQuest and the old SoundDiver only really provide a veneer of control over the synths they support, often implemented in a very inefficient manner, and often with serious bugs that they won't fix.

**Reply by unityconstruct** [reply]
haha.. going to assume they do that 'bit packing' absurdity that makes the trivial, monumental
In addition to be an equipment hoarder... I have a similar dysfunction with saving EVERYTHING i come across... 

> Companies like Soundtower, SoundQuest and the old SoundDiver only really provide a veneer of control over the synths they support, often implemented in a very inefficient manner, and often with serious bugs that they won't fix.

@Godlike-Productions

**Reply by Godlike-Productions** [reply]
Don't get me started on bit packing.  Everyone does it differently.  And then there are checksums that evaluate to negative numbers without disclosing their algorithms (side eye at Eventide)

Don't get me wrong. Kurzweil's are amazing and I'm grateful at their long term support.  They synths internally are bulletproof, and flexible.  It's us trying to do stuff they were never designed for that is the cause, and the turnover of engineers and lack of code comments.

EMU do have some nice filters though.  I have an ESI32 in my studio, and an Orbit, but honestly rarely use them. We are really spoilt for choice though and Kurzweil's just work from power on without having to load any samples.

**Comment by unityconstruct** [comment, marked as answer]
### Old Official Builds: https://ctrlr.org/nightly/?C=M;O=D
[Ctrlr-5.3.191.exe](https://ctrlr.org/nightly/Ctrlr-5.3.191.exe)                                  07-Apr-2017 11:31            12418210
[Ctrlr-5.3.193.exe](https://ctrlr.org/nightly/Ctrlr-5.3.193.exe)                                  07-Apr-2017 11:31            12415654
[Ctrlr-5.3.196.exe](https://ctrlr.org/nightly/Ctrlr-5.3.196.exe)                                  07-Apr-2017 11:31            12417540
> _**[Ctrlr-5.3.198.dmg](https://ctrlr.org/nightly/Ctrlr-5.3.198.dmg)                                  07-Apr-2017 11:33            45063483**_

[Ctrlr-5.3.199.exe](https://ctrlr.org/nightly/Ctrlr-5.3.199.exe)                                  07-Apr-2017 11:34            12446672

> _**[Ctrlr-5.3.201.exe](https://ctrlr.org/nightly/Ctrlr-5.3.201.exe)                                  07-Apr-2017 11:35            12452001**_
- in the ballpark with 5.3.201 in codebase, but will be tricky if 5.3 was updated after these builds
- assuming 5.3.201 is build 201 - meaning that 201 builds done on a particular commit..
  - highly doubtful 201 will be found anywhere in repo, so inference is really only way
  - given multiple builds can occur in a day OR no builds can occur in a day.. mostly a commt-by-commit check
  - there may be NO benefit to anything beyond the binaries, so pausing search ATM
- FILEVERSION  5,3,0,0
- [9a2ce62](https://github.com/RomanKubiak/ctrlr/commit/9a2ce62201bcc03033edb6f03b00930405897265)
- 6 years ago 6/18/2018

[Ctrlr-5.4.11.exe](https://ctrlr.org/nightly/Ctrlr-5.4.11.exe)                                   07-Apr-2017 11:37            12950017
[Ctrlr-5.4.16.exe](https://ctrlr.org/nightly/Ctrlr-5.4.16.exe)                                   07-Apr-2017 11:38            14225589
[Ctrlr-5.4.2.exe](https://ctrlr.org/nightly/Ctrlr-5.4.2.exe)                                    07-Apr-2017 11:35            12949582
[Ctrlr-5.4.24.exe](https://ctrlr.org/nightly/Ctrlr-5.4.24.exe)                                   07-Apr-2017 11:39            14857617
[Ctrlr-5.4.27.exe](https://ctrlr.org/nightly/Ctrlr-5.4.27.exe)                                   07-Apr-2017 11:40            14953142
[Ctrlr-5.4.29.exe](https://ctrlr.org/nightly/Ctrlr-5.4.29.exe)                                   07-Apr-2017 11:40            14869130
[Ctrlr-5.5.2.exe](https://ctrlr.org/nightly/Ctrlr-5.5.2.exe)                                    08-May-2017 15:45            18490878
[Ctrlr-6.0.1.exe](https://ctrlr.org/nightly/Ctrlr-6.0.1.exe)                                    09-Jan-2019 20:03            25789966
[Ctrlr-6.0.1.win32.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.1.win32.build.log)                        09-Jan-2019 20:03                   0
[Ctrlr-6.0.1.win64.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.1.win64.build.log)                        09-Jan-2019 20:03                   0
[Ctrlr-6.0.25.exe](https://ctrlr.org/nightly/Ctrlr-6.0.25.exe)                                   03-Oct-2019 12:45            25738846
[Ctrlr-6.0.25.win32.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.25.win32.build.log)                       03-Oct-2019 12:45                   0
[Ctrlr-6.0.25.win64.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.25.win64.build.log)                       03-Oct-2019 12:45                   0

> _**[Ctrlr-6.0.26.exe](https://ctrlr.org/nightly/Ctrlr-6.0.26.exe)                                   07-Oct-2019 14:54            31118337**_
> _**[Ctrlr-6.0.26.win32.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.26.win32.build.log)                       07-Oct-2019 14:54                   0**_
> _**[Ctrlr-6.0.26.win64.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.26.win64.build.log)                       07-Oct-2019 14:54                   0**_

[Ctrlr-6.0.4.exe](https://ctrlr.org/nightly/Ctrlr-6.0.4.exe)                                    16-Jan-2019 17:33            26294401
[Ctrlr-6.0.4.win32.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.4.win32.build.log)                        16-Jan-2019 17:33                   0
[Ctrlr-6.0.4.win64.build.log](https://ctrlr.org/nightly/Ctrlr-6.0.4.win64.build.log)                        16-Jan-2019 17:33                   0
[Ctrlr-i686-5.3.189.sh](https://ctrlr.org/nightly/Ctrlr-i686-5.3.189.sh)                              07-Apr-2017 11:30            13998261
[Ctrlr-i686-5.3.191.sh](https://ctrlr.org/nightly/Ctrlr-i686-5.3.191.sh)                              07-Apr-2017 11:30            13988857
[Ctrlr-i686-5.3.199.sh](https://ctrlr.org/nightly/Ctrlr-i686-5.3.199.sh)                              07-Apr-2017 11:34            14013029
[Ctrlr-i686-5.4.16.sh](https://ctrlr.org/nightly/Ctrlr-i686-5.4.16.sh)                               07-Apr-2017 11:39            14232884
[Ctrlr-x86_64-5.3.188.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.3.188.sh)                            07-Apr-2017 11:29            18170479
[Ctrlr-x86_64-5.3.192.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.3.192.sh)                            07-Apr-2017 11:30            18123438
[Ctrlr-x86_64-5.3.196.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.3.196.sh)                            07-Apr-2017 11:32            18129802
[Ctrlr-x86_64-5.3.199.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.3.199.sh)                            07-Apr-2017 11:34            18208405

> _**[Ctrlr-x86_64-5.3.201.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.3.201.sh)                            07-Apr-2017 11:35            18213576**_

[Ctrlr-x86_64-5.4.11.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.4.11.sh)                             07-Apr-2017 11:36            16680052
[Ctrlr-x86_64-5.4.14.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.4.14.sh)                             07-Apr-2017 11:37            16615885
[Ctrlr-x86_64-5.4.16.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-5.4.16.sh)                             07-Apr-2017 11:38            16617332
[Ctrlr-x86_64-6.0.0.linux.build.log](https://ctrlr.org/nightly/Ctrlr-x86_64-6.0.0.linux.build.log)                 10-Jan-2019 16:22                 314
[Ctrlr-x86_64-6.0.0.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-6.0.0.sh)                              10-Jan-2019 16:22            16504841
[Ctrlr-x86_64-6.0.1.linux.build.log](https://ctrlr.org/nightly/Ctrlr-x86_64-6.0.1.linux.build.log)                 10-Jan-2019 16:25                 314
[Ctrlr-x86_64-6.0.1.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-6.0.1.sh)                              10-Jan-2019 16:25            16503899
[Ctrlr-x86_64-6.0.35.linux.build.log](https://ctrlr.org/nightly/Ctrlr-x86_64-6.0.35.linux.build.log)                11-May-2020 18:27                 501
[Ctrlr-x86_64-6.0.35.sh](https://ctrlr.org/nightly/Ctrlr-x86_64-6.0.35.sh)

**Comment by damiensellier** [comment]
Hi @unityconstruct , you're spot on.
Most people working with VST2.4 only and standalone usually stick to 5.3.198 for mac and 5.3.201 for PC.
Releasing VST2 plugins requires having the old SDK license agreement with steinberg.


For anyone releasing VST3 the only way to go is with 5.6.23 and above.
Releasing VST3 requires the current VST3 license agreement from Steinberg, it's just a matter of sending a signed pdf to them and you'll receive it countersigned a few days after and you'll be good to go.
https://steinbergmedia.github.io/vst3_dev_portal/resources/VST3_License_Agreement.pdf

VST3 builds require at least the Ctrlr version moddified by @jmechnich to work properly with xcode. 
https://github.com/jmechnich/ctrlr/releases/tag/5.6-f452347

My version CtrlrX base on @jmechnich 5.6.23 is currently at v5.6.28 and has new features and bug fixes.  The changelog is on the fork home page.
https://github.com/damiensellier/CtrlrX

Enjoy!

**Reply by unityconstruct** [reply]
Hi @damiensellier  I stole your notes & added above, but failed to pass on THANKS!... Haven't built your latest `29`, but WHOA! Looks like you drug Ctrlr out of the 80's sir! 

## Hatz off.

**Reply by damiensellier** [reply]
Thanks for your different version Logs, it helps understanding where we are at. 
Roman Kubiak who is the creator of Ctrlr gave up the project a few years ago, probably around 2018.

Everyone here on this github are all versed into synths, external hardware FX and it seems we are more into doing music... but for our own need we had to keep the project (almost) up to date to be able to design panels for our beloved MIDI controlled gears.

My fork called CtrlrX is relying on Jorg's @jmechnich Ctrlr spin. If your project is working with 5.6.23,  99.99% chance it will work with CtrlrX. There's no major updates in there and I will keep JUCE 6.0.8 as long as it's possible.
JUCE 7 just brought ARA which we don't need with a few audio processing fixes and JUCE 8 will offer new graphic rendering options for modern workstations as well as a better integration for the iLok protection. So JUCE 6 is still a solid version of the library for our needs. 

You can export VST3 "intermediates" as I call them to build properly named and identified plugins with CtrlrX. It works for both mac Intel and M1 M2 and PC. I'm not into Linux so I can't tell if it's working on this particular platform. I started working on the AAX version of CtrlrX for pro tools, but Avid's partner prohibitive requirements ($$$) prevented me from going further.  I had a dev account but it needs to be renewed every year. It's far from being enticing and hated that "exclusive" approach. Most people using ProTools have VST wrappers anyway.

CtrlrX will stay at version 5.6.xx until we have every bug/issue fixed, especially the VST3 naming issue. It has to export instances properly to deliver VST3 plugins with juce one click. It's unfortunately not the case right now. I always post version updates on the Ctrlr discussion section to let people know about the new features, and the changelog is always updated on the .md of my fork.

Thanks for your work. Welcome to the community, hope you'll design nice panels with those tools!
Damien

**Reply by unityconstruct** [reply]
> hope you'll design nice panels with those tools!

### lol... that makes 2 of us....! 

Anyway your recent UI update is SLLLLIIIIICKKK sir!

**Reply by unityconstruct** [reply]
last year's stab at the Prodatum UI port.. with a new UI mockup attempt built while traversing that HELPFUL StepByStep doc
![image](https://github.com/RomanKubiak/ctrlr/assets/40268430/b9e45180-0b24-413d-9620-6c8a256e9644)

**Comment by Godlike-Productions** [comment]
One other consideration that's important to me is the iLok linker code for OSX to support PACE copy protection.

One other thing I have noticed with VST3's built from the original 5.6.23 is there is an issue with the canvas and panel size in Windows.  Has this been sorted in your latest CtrlrX version?  I'm looking at doing some Windows VST3 builds again, and looking at the best current version to compile?

I applaud all the development going on, I'm just struggling with which version is the best for VST3's right now.

**Reply by damiensellier** [reply]
Hi, the xlinker to give the extra space required for ilok is always added to my jucer file in the xcode mac settings :
**-Xlinker -headerpad -Xlinker 800**

![xlinker](https://github.com/RomanKubiak/ctrlr/assets/55164385/8b451dc6-671f-4d4c-a040-d89e87f537c5)


The canvas and panel size (showing scrollers) have been fixed in my version 5.6.28. https://github.com/RomanKubiak/ctrlr/discussions/588. We talked together last year about being able to design responsive/scalable panels, instead of forcing different size settings... it's now available with v5.5.28+ (More details here : https://github.com/RomanKubiak/ctrlr/discussions/588). Please Confirm this is what you're talking about.
I released new version 5.6.29 with new features related to the GUI lookandfeel_v4, it's in test right now, it's working on mac intel and needs further testing on other port.

**Reply by unityconstruct** [reply]
- @damiensellier  - trapped notes & updated `CtrlrX-5.6.28` to `CtrlrX-5.6.29` at top
- @Godlike-Productions - lol... I feel your pain brother! Thus the attempt at a decoder ring up top.
  - hopefully its accurate
  - Assuming `5.6.29` would be an ideal release if touching code, but I'm still the village idiot on the ecosystem. Based on Panel target, there were specific nuances with each version. I think you've been on the scene a while so hopefully I'm not re-stating the obvious.
  - if something is grotesquely wrong lmk & I can massage the tables at top.
  - will eventually further distill the info once I stop running into relevant posts on versions.
- I picked up some context from Ted-juh a week or so ago, that helped delineate.
  - https://github.com/RomanKubiak/ctrlr/discussions/599#discussioncomment-8261168
>[Ted-juh](https://github.com/Ted-juh)
>[2 weeks ago](https://github.com/RomanKubiak/ctrlr/discussions/599#discussioncomment-8261168)
>While Ctrlr 5.3.201 is the "stable" version, I can only advice to aim at version 5.6.23. Vst2 is out of the window, Juce was updated to version 6 (prior version 5) and meant a huge overhaul for Ctrlr which Roman was, thankfully, providing us.
>All changes made ( mouse events, vst3 name change support, and so on) will be made on the newer version anyway.
>See Damien's fork CtlrX. Some excellent additions there.
```

**Reply by bijlevel** [reply]
@unityconstruct Great overview, must have cost you an awful lot of time! I have some additions to your overview. I have 5.6.23 to 5.6.29 compiled for **Silicon Mac** succesfully using **Juce 6.1.4**. On this Silicon Mac also the AU and VST3 do load OK in my DAW's (Digital Performer 11 and Mixbus 9 for Silicon Mac). I also compiled succesfully these CTRLR (Standalone, VST3) versions for Win 10 and Win 11, but both Windows systems are running within VMWare Fusion, because I don't have a PC. But I'm confident these CTRLR-version will also run on a "real" PC. I did look at the differences between Juce 6.0.8 and 6.1.4. and it seems mainly being about better VST3 support. So I propose to everyone to try and build CTRLR with Juce 6.1.4, because it may fix VST3 issues?

**Reply by damiensellier** [reply]
Hi @bijlevel , thanks for your feedback, can you please post individual zip files of your Apple silicon of 5.6.29, standalone, VST3 and AU builds on the discussion https://github.com/RomanKubiak/ctrlr/discussions/604 ? We would get all the available versions over there that would be awesome! Thanks! Damien

PS: I'll upgrade to Juce 6.1.4 for the next release, it'sa nice suggestion.
